pkg/bmh/baremetalhost.go

```diff
@@ -451,32 +436,37 @@ func (builder *BmhBuilder) Exists() bool {
 	glog.V(100).Infof("Checking if baremetalhost %s exists in namespace %s",
 		builder.Definition.Name, builder.Definition.Namespace)
 
-	var err error
-	builder.Object, err = builder.Get()
+	err := builder.Discover()
 
 	return err == nil || !k8serrors.IsNotFound(err)
 }
 
-// Get returns bmh object if found.
-func (builder *BmhBuilder) Get() (*bmhv1alpha1.BareMetalHost, error) {
```

Comment:
Please restore the Get method. Metal3 does not provide a versioned client interface that provides a Get method